swap/swap_test.go

```diff
@@ -184,267 +184,197 @@ func testBalances(t *testing.T, swap *Swap, expectedBalances map[enode.ID]int64)
 	}
 }
 
-// TestSentCheque verifies that sent cheques data is correctly obtained
-func TestSentCheque(t *testing.T) {
+// TestCheques verifies that sent and received cheques data for all known swap peers is correct
+func TestCheques(t *testing.T) {
 	// create a test swap account
 	swap, clean := newTestSwap(t, ownerKey, nil)
 	defer clean()
 
-	// test cheque addition for peer
-	testPeer, err := swap.addPeer(newDummyPeer().Peer, common.Address{}, common.Address{})
-	if err != nil {
-		t.Fatal(err)
-	}
+	// check cheques are empty
+	testChequesByPeerAndType(t, swap, map[enode.ID]map[string]*Cheque{})
 
-	sentCheque, err := swap.SentCheque(testPeer.ID())
-	if err != nil {
-		t.Fatal(err)
-	}
-	if sentCheque != nil {
-		t.Fatalf("Expected sent cheque to be nil, but is %v", sentCheque)
-	}
+	// add peer
+	testPeer := addPeer(t, swap)
+	testPeerID := testPeer.ID()
 
-	// generate a random cheque as sent
-	generatedCheque := newRandomTestCheque()
-	err = testPeer.setLastSentCheque(generatedCheque)
-	if err != nil {
-		t.Fatal(err)
-	}
-	sentCheque, err = swap.SentCheque(testPeer.ID())
-	if err != nil {
-		t.Fatal(err)
-	}
-	if sentCheque != generatedCheque {
-		t.Fatalf("Expected sent cheque to be %v, but is %v", generatedCheque, sentCheque)
-	}
+	// test no cheques are present
+	testChequesByPeerAndType(t, swap, map[enode.ID]map[string]*Cheque{testPeerID: {receivedChequeResponseKey: nil, sentChequeResponseKey: nil}})
```

Comment:
this is really not a clean approach and very difficult to maintain and review. please consider to migrate this test to a table driven test

Reply:
i agree that it's quite hard to read. i am not sure how i can apply table driven tests here though, given that the tests are sequential: check returned cheques after starting, then after adding a peer, then after setting a sent cheque, then a received one, a second peer, and so on. 

do you have any suggestions or examples on how to combine this with a table-driven approach?

Reply:
well, in this test you're basically doing:
1. setup
2. create new cheque and setting it
3. check the expected values in the map
4. repeat 2,3
5. ..


you can create a table driven test with a struct that contains the expectations of the values you'd be looking for.
i am also not sure why you need multiple peers in this scenario. does this really improve the code coverage?

Reply:
>i am also not sure why you need multiple peers in this scenario. does this really improve the code coverage?

it does not improve code coverage. but given that the end result is a map which includes all cheques per peer, it's important to test that the result is still correct when working with _at least_ 2 of them, even if it does not affect coverage (and it does not).

>you can create a table driven test with a struct that contains the expectations of the values you'd be looking for.

i've uploaded the `TestChequesNew` function, which would replace `TestCheques` if deemed preferable. 

this my attempt at implementing an equivalent test in table-driven form. should it be accepted, it would do away with the `addPeer`, `setNewCheque` and `saveNewCheque` functions too.

i don't find this implementation particularly easier to read or maintain. let me know if you do.

Reply:
retired old tests and auxiliary functions